qiskit/transpiler/passmanager.py

```diff
@@ -312,3 +312,179 @@ def passes(self) -> List[Dict[str, BasePass]]:
                 item["flow_controllers"] = {}
             ret.append(item)
         return ret
+
+
+class FullPassManager(PassManager):
+    """A full Pass manager pipeline for a backend
+
+    Instances of FullPassManager define a full compilation pipeline from a abstract virtual
+    circuit to one that is optimized and capable of running on the specified backend. It is
+    built using predefined stages:
+
+    1. Init - any initial passes that are run before we start embedding the circuit to the backend
+    2. Layout - This stage runs layout and maps the virtual qubits in the
+        circuit to the physical qubits on a backend
+    3. Routing - This stage runs after a layout has been run and will insert any
+        necessary gates to move the qubit states around until it can be run on
+        backend's compuling map.
+    4. Translation - Perform the basis gate translation, in other words translate the gates
+        in the circuit to the target backend's basis set
+    5. Pre-Optimization - Any passes to run before the main optimization loop
```

Comment:
I was going to add hook points in a follow up PR when I looked at adding arguments to transpile. Especially since it confuses things because the backend hook point methods I was thinking of were things like `pre_routing` and `post_routing` which confuses things a bit if it's backend specified and user specified.

The pre and post optimization stages here were what I was doing to try and account for passes that ran after basis translation but before the optimization loop or passes after the optimization loop in the preset pass managers (when I originally wrote this 4 months ago).

But coming back to this with fresh eyes now, I agree it's probably better to either leave them as strictly modifiers/hook points or explicit define all the in between stages.